src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -218,7 +217,7 @@ public static bool IsValidEnumType(this TypeSymbol type)
         {
             var underlyingType = type.GetEnumUnderlyingType();
             // SpecialType will be None if the underlying type is invalid.
-            return (underlyingType is object) && (underlyingType.SpecialType != SpecialType.None);
+            return underlyingType is { SpecialType: >= SpecialType.System_Object and <= SpecialType.System_Runtime_CompilerServices_InlineArrayAttribute };
```

Comment:
Got to here today, I'll try to pick it up again Monday or Tuesday. Looking good overall so far, with just a couple of minor comments/questions.